File: testdata/bench/word_count.js
var text = "";
var n = 20000;

for (var i = 0; i < n; i++) {
    text += "lorem ipsum dolor sit amet consectetur adipiscing elit ";
}

var words = text.split(" ");
var counts = {};

for (var i = 0; i < words.length; i++) {
    var w = words[i];
    if (w === "") {
        continue;
    }
    if (counts[w] === undefined) {
        counts[w] = 1;
    } else {
        counts[w] += 1;
    }
}

var out = 0;
for (var k in counts) {
    out += counts[k];
}

console.log(out);

File: testdata/bench/word_count.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	n := 20000
	var sb strings.Builder
	for i := 0; i < n; i++ {
		sb.WriteString("lorem ipsum dolor sit amet consectetur adipiscing elit ")
	}
	text := sb.String()

	words := strings.Split(text, " ")
	counts := make(map[string]int)
	for _, w := range words {
		if w == "" {
			continue
		}
		counts[w]++
	}

	out := 0
	for _, v := range counts {
		out += v
	}
	fmt.Println(out)
}
